### PR TITLE
Fix a possible recursion in SSLfatal handling

### PR DESCRIPTION
**Pull request by bernd-edlinger**
Fixes: #7161 (hopefully)

### CODE REVIEW

**Comment by bernd-edlinger** [comment]
This might be a candidate for the upcoming release version.
In the debug version the assertion will prevent the recursion,
but in the release version NDEBUG is defined, and kills the assert.

**Comment by mattcaswell** [comment]
Too late I think.

**Comment by mattcaswell** [comment]
(we just completed our pre-release checks - don't want to have to restart them)

**Comment by bernd-edlinger** [comment]
Merged to 1.1.1 and master branch.  Thanks!

File: ssl/statem/statem.c [hunk 1]
```diff
@@ -118,11 +118,12 @@ void ossl_statem_set_renegotiate(SSL *s)
 void ossl_statem_fatal(SSL *s, int al, int func, int reason, const char *file,
                        int line)
 {
+    ERR_put_error(ERR_LIB_SSL, func, reason, file, line);
     /* We shouldn't call SSLfatal() twice. Once is enough */
-    assert(s->statem.state != MSG_FLOW_ERROR);
+    if (s->statem.in_init && s->statem.state == MSG_FLOW_ERROR)
+      return;
```
Comment (bernd-edlinger) [inline ssl/statem/statem.c]: Oops, I have accidentally used 2-spaces GCC-indentation style...

**Pull request closed without merging**